sdk/src/main/java/io/opentelemetry/sdk/resources/ResourceConstants.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2020, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.sdk.resources;
+
+/**
+ * Provides constants for resource semantic conventions defined by the OpenTelemetry specification.
+ *
+ * @see <a
+ *     href="https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/data-resource-semantic-conventions.md">Resource
+ *     Conventions</a>
+ */
+public class ResourceConstants {
```

Comment:
Please split this in a separate PR :)

Reply:
Also we should consider if using am xml, yaml file for all the constants is appropriated and share that between libraries.

Reply:
should these semantic conventions be encoded in some protobuf definitions, so they can be shared across all the language implementations?

Reply:
I checked the projects. opentelemetry-go, opentelemetry-collector, opentelemetry-dotnet, opentelemetry-auto-instr-java, opentelemetry-js all have constants for semantic conventions defined already. I can see the benefit of having one file to update but not sure it is workable.

I will put the constants in a separate PR. Should I put them in the API where they are available to users? It seems to make sense given they are part of the specification.

Reply:
They are part of the SDK specification (resource is part of the SDK). The attributes for trace are part of the API and for those we probably need them in the API.